src/test/java/org/cactoos/http/HtHeadTest.java

```diff
@@ -107,4 +109,138 @@ public void largeText() throws IOException {
         );
     }
 
+    @Test
+    public void attackInTheMiddle() throws IOException {
+        final int size = 16384;
+        final ByteBuffer buf = ByteBuffer.allocate(size * 2);
+        for (int pos = 0; pos < size - 4; ++pos) {
+            buf.put((byte) 'a');
+        }
+        buf.put((byte) 'b');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        for (int pos = buf.position(); pos < 2 * size; ++pos) {
+            buf.put((byte) 'a');
+        }
+        MatcherAssert.assertThat(
+            "Can't find when separator broken by two buffers",
+            new TextOf(
+                new HtHead(
+                    new InputOf(
+                        new TextOf(new BytesOf(buf.array())).asString()
+                    )
+                )
+            ).asString(),
+            Matchers.endsWith("b")
+        );
+    }
+
+    @Test
+    public void attackOnBufferStart() throws IOException {
+        final int size = 16384;
+        final ByteBuffer buf = ByteBuffer.allocate(size * 2);
+        for (int pos = 0; pos < size - 1; ++pos) {
+            buf.put((byte) 'a');
+        }
+        buf.put((byte) 'b');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        for (int pos = buf.position(); pos < 2 * size; ++pos) {
+            buf.put((byte) 'a');
+        }
+        MatcherAssert.assertThat(
+            "Can't find when buffers starts from separator",
+            new TextOf(
+                new HtHead(
+                    new InputOf(
+                        new TextOf(new BytesOf(buf.array())).asString()
+                    )
+                )
+            ).asString(),
+            Matchers.endsWith("b")
+        );
+    }
+
+    @Test
+    public void fakeSeparator() throws IOException {
+        final int size = 16384;
+        final ByteBuffer buf = ByteBuffer.allocate(size * 2);
+        for (int pos = 0; pos < size - 2; ++pos) {
+            buf.put((byte) 'a');
+        }
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        for (int pos = size; pos < 2 * size - 5; ++pos) {
+            buf.put((byte) 'a');
+        }
+        buf.put((byte) 'b');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        buf.put((byte) '\r');
+        buf.put((byte) '\n');
+        MatcherAssert.assertThat(
+            "Can't add previous buffer with part of separator",
+            new TextOf(
+                new HtHead(
+                    new InputOf(
+                        new TextOf(new BytesOf(buf.array())).asString()
+                    )
+                )
+            ).asString(),
+            new MatcherOf<>(
+                str -> str.length() == size * 2 - 4 && str.endsWith("b")
+            )
+        );
+    }
+
+    @Test
+    public void rollbackState() throws IOException {
```

Comment:
@SharplEr what is this test doing?

Reply:
@llorllale just increases code coverage to an acceptable level.

Reply:
@SharplEr the test must be testing *something*. Let's make its name more descriptive please